src/github.com/getlantern/fronted/direct.go

```diff
@@ -136,6 +137,8 @@ func (d *direct) Do(req *http.Request) (*http.Response, error) {
 // Dial persistently dials masquerades until one succeeds.
 func (d *direct) Dial(network, addr string) (net.Conn, error) {
 	gotFirst := false
+	candidateCh := <-candidateChCh
+	candidateChCh <- candidateCh
```

Comment:
> BTW, the problem with putting candidateCh back on the candidateChCh is that you'll end up with two versions of candidateCh on candidateChCh after Configure() is called a second time (because candidateChCh is sized to 1). After a 3rd call to Configure(), either Configure() or Dial() will get deadlocked because the channel is full at that point.

Yeah that's what I was saying about that `candidateChCh <- candidateCh` line blocking.

The big problem with the atomic.Value, though, goes back to the original race issue, i.e. if `Dial` reads it before `Configure` is called, then `candidateCh` will be empty and never populated (it will be replaced with a new one when `Configure` is called, but `Dial` will be none the wiser).

I just committed your select change. I like that a little better than the straight blocking channel because AFAICT in the case of multiple Configure calls we'd be building up a bunch of chans in the queue that we will never need (since we only care about the latest from `Configure`) as well as building up a bunch of go routines with a goroutine call that executes `candidateChCh <- candidateCh`

Anyway, what do you think of the code now?